src/bootstrap/setup.rs

```diff
@@ -68,11 +68,17 @@ pub fn setup(src_path: &Path, profile: Profile) {
 
     let path = cfg_file.unwrap_or_else(|| src_path.join("config.toml"));
     let settings = format!(
-        "# Includes one of the default files in src/bootstrap/defaults\n\
-    profile = \"{}\"\n",
+        r#"# Includes one of the default files in src/bootstrap/defaults
+profile = "{}"
+
+[llvm]
+# Will download LLVM from CI if available on your platform (Linux only for now)
+# https://github.com/rust-lang/rust/issues/77084 tracks support for more platforms
+download-ci-llvm = "if-available"
```

Comment:
Could we just duplicate this across the actual default files? I would prefer to avoid adding even more layers of places where defaults are getting set.

Reply:
Yeah, in my mind `setup.rs` is for setting defaults that are _not_ the same for everyone, while `src/bootstrap/defaults` are for things that can reasonably be shared.

Reply:
Done! Added to all default configs. I just have a doubt for the `config.codegen.toml`, maybe people working close to codegen don't want the CI built one by default?

Reply:
Ah, it seems moving these to the default profiles broke the bootstrap, because bootstrap.py doesn't know about the `profile` setting (I'm guessing it's only handled in the `config.rs` file)

Reply:
Looking further into this, it seems bootstrap only has a basic regex mechanism to lookup toml keys, and returns the first key that matches, which means appending the contents of our default file *should* work as expected, allowing the main config.toml to override these defaults if needed. Definitely feels like a hack though :D I'll push some code soon

Reply:
What do you mean by broke bootstrap? rustbuild should be built before LLVM IIRC.

> Done! Added to all default configs. I just have a doubt for the config.codegen.toml, maybe people working close to codegen don't want the CI built one by default?

Oops, good catch, I'm glad we moved this to config files - it should only show up for the `library` and `compiler` profiles, not anything else.

Reply:
Fixed, removed the llvm stuff from `user` and `codegen` :)

Reply:
I've pushed a fix to load the included config in the bootstrap.py as well. Let me know if you think this is a viable solution, and I'll polish it / add some comments

Reply:
> What do you mean by broke bootstrap? rustbuild should be built before LLVM IIRC.

Running this:
```
./x.py setup compiler
./x.py build
```

was throwing this error:

```
Updating only changed submodules
Submodules updated in 0.02 seconds
    Finished dev [unoptimized + debuginfo] target(s) in 0.11s


failed to execute command: "/home/alarsyo/work/rust/build/x86_64-unknown-linux-gnu/ci-llvm/bin/llvm-config" "--libfiles"
error: No such file or directory (os error 2)


failed to run: /home/alarsyo/work/rust/build/bootstrap/debug/bootstrap build
Build completed unsuccessfully in 0:00:00
```

Because bootstrap.py is responsible for downloading the LLVM build from CI, and that doesn't happen since it doesn't see options from the default file included with `profile = "compiler"`

Reply:
Got it, thanks.

@Mark-Simulacrum how hard would it be to move that into rustbuild instead of bootstrap.py? Doesn't need to block this, but I'd like to do as little as possible in python anyway.

Reply:
That error comes from the Rust portion of rustbuild, I think? It looks like a Rust error...

Presumably LLVM was not downloaded or not unpacked which led to this problem.

Reply:
FWIW, I'd be interested in tackling this to get more familiar with the bootstrap code, if you decide to move this to rustbuild :)

Reply:
Let's tackle this in a follow-up, it seems reasonable to have bootstrap.py read profiles anyway.